### PR TITLE
Create suggestions

### LINKED ISSUE #65: [DEV] As a user, I want to have suggestion on my typos

### PR DESCRIPTION
**Pull request by ngsengleng**
### Summary

Resolves #65 

This PR adds functionality and test cases for displaying top 5 most similar commands if the user provides an invalid command.

### Detail

The function is created within commons/Messages.java but i dont know if this is the right place to park it and the list of commands is also hard coded into the function as it seems to be the best way to do it for now. Will look into making it better so that it only displays simliar commands if the invalid command isn't completely gibberish.

### Pictures / Gif

_provide gif or screenshot of the features if there is any_
Screenshot 2021-10-06 at 2 44 00 AM
 **[requires scrolling within GUI]**
### Checks

- [x] Link PR to issue
- [x] Add Reviewer
- [x] Pass CI

### CODE REVIEW

File: build.gradle
Comment (simonjulianl) [inline build.gradle:47]: I think we might need to get permission from prof using the forum for using additional external library like this

File: src/test/java/gomedic/commons/core/MessagesTest.java
Comment (simonjulianl) [inline src/test/java/gomedic/commons/core/MessagesTest.java:13]: Actually, whats the max distance that you have set for the command, seems that clear and addle is like 5 distance apart but seems still included here

**Review by simonjulianl — CHANGES_REQUESTED** [review]
LGMT, except for the distance threshold (seems here there is no limit, its just take top 5 smallest distance but still there should like threshold like 2 chars distance, otherwise clear typo would suggest delete ) and the permission to use the external library (otherwise, it's pretty easy to implement one also)

File: build.gradle
Comment (ngsengleng) [reply build.gradle:47]: yep ive submitted a request, waiting for approval

File: src/test/java/gomedic/commons/core/MessagesTest.java
Comment (ngsengleng) [reply src/test/java/gomedic/commons/core/MessagesTest.java:13]: i dont have a max distance right now, but I am considering setting a threshold of maybe string.length / 2 to filter out the unwanted commands

File: build.gradle
Comment (ngsengleng) [reply build.gradle:47]: ok it has been approved

Comment (simonjulianl) [reply build.gradle:47]: thank you man

File: src/test/java/gomedic/commons/core/MessagesTest.java
Comment (simonjulianl) [reply src/test/java/gomedic/commons/core/MessagesTest.java:13]: I think its possible

**Review by simonjulianl — CHANGES_REQUESTED** [review]
After changing the max distance threshold, it should be fine also the resolve keyword only works in the format Resolves #<number> , so yup can always check the linked issue

**Review by simonjulianl — APPROVED** [review]
LGTM

**Comment by simonjulianl** [comment]
btw, for the check, maybe can try to make the checks pass first (I see some checkstyle issue)

**Comment by ngsengleng** [comment]
Suggestions is implemented, but i think the threshold for closest valid commands can be finetuned if issues arise in the future

**Pull request merged**
